wiki/plugins/attachments/migrations/0001_initial.py

```diff
@@ -0,0 +1,78 @@
+                ('automatic_log', models.TextField(editable=False, blank=True)),
+                ('ip_address', models.GenericIPAddressField(verbose_name='IP address', null=True, editable=False, blank=True)),
+                ('modified', models.DateTimeField(auto_now=True)),
+                ('created', models.DateTimeField(auto_now_add=True)),
+                ('deleted', models.BooleanField(default=False, verbose_name='deleted')),
+                ('locked', models.BooleanField(default=False, verbose_name='locked')),
+                ('file', models.FileField(upload_to=wiki.plugins.attachments.models.upload_path, verbose_name='file')),
+                ('description', models.TextField(blank=True)),
+            ],
+            options={
+                'ordering': ('created',),
+                'get_latest_by': 'revision_number',
+                'verbose_name': 'attachment revision',
+                'verbose_name_plural': 'attachment revisions',
+                'db_table': 'wiki_attachments_attachmentrevision',
+            },
+            bases=(models.Model,),
+        ),
+        migrations.CreateModel(
+            name='Attachment',
+            fields=[
+                ('reusableplugin_ptr', models.OneToOneField(parent_link=True, auto_created=True, primary_key=True, serialize=False, to='wiki.ReusablePlugin')),
+                ('original_filename', models.CharField(max_length=256, null=True, verbose_name='original filename', blank=True)),
+            ],
+            options={
+                'verbose_name': 'attachment',
+                'verbose_name_plural': 'attachments',
+                'db_table': 'wiki_attachments_attachment',
+            },
+            bases=('wiki.reusableplugin',),
+        ),
+        migrations.AddField(
+            model_name='attachmentrevision',
+            name='attachment',
+            field=models.ForeignKey(to='wiki_attachments.Attachment'),
+            preserve_default=True,
+        ),
+        migrations.AddField(
+            model_name='attachmentrevision',
+            name='previous_revision',
+            field=models.ForeignKey(blank=True, to='wiki_attachments.AttachmentRevision', null=True),
+            preserve_default=True,
+        ),
+        migrations.AddField(
+            model_name='attachmentrevision',
+            name='user',
+            field=models.ForeignKey(verbose_name='user', blank=True, to=settings.AUTH_USER_MODEL, null=True),
+            preserve_default=True,
+        ),
+        migrations.AddField(
+            model_name='attachment',
+            name='current_revision',
+            field=models.OneToOneField(related_name='current_set', null=True, to='wiki_attachments.AttachmentRevision', blank=True, help_text='The revision of this attachment currently in use (on all articles using the attachment)', verbose_name='current revision'),
+            preserve_default=True,
+        ),
+    ]
```

Comment:
Why the separate`AddField`s after the `CreateModel`s? Is this just several migrations from upstream squashed together?

Reply:
This is what was in our original `001_initial` migration for the wiki app, copied over verbatim.  This was a particularly tricky case because `AttachmentRevision` has a foreign key to itself, plus it and `Attachment` have foreign keys to each other.  I think Django just erred on the side of caution and added each of the foreign keys individually after creating the tables.